sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/Models/TimeSeriesId.cs

```diff
@@ -0,0 +1,157 @@
+    /// </summary>
+    /// <remarks>
+    /// A single Time Series Id can be composite of multiple properties are specified as Time Series Id at environment creation time.
+    /// The position and type of values must match Time Series Id properties specified on the environment and returned by Get Model Setting API.
+    /// </remarks>
+    public class TimeSeriesId
+    {
+        private (bool, object) _key1;
+        private (bool, object) _key2;
+        private (bool, object) _key3;
+
+        internal (bool, object) Key1 => _key1;
+
+        internal (bool, object) Key2 => _key2;
+
+        internal (bool, object) Key3 => _key3;
+
+        /// <summary>
+        /// Creates a new Time Series Id with 1 key.
+        /// </summary>
+        /// <param name="key1">The first key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 key.
+        /// </remarks>
+        public TimeSeriesId(object key1)
+        {
+            _key1 = (true, key1);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 2 keys.
+        /// </summary>
+        /// <param name="key1">The first key that identifies a Time Series instance.</param>
+        /// <param name="key2">The second key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 2 keys.
+        /// </remarks>
+        public TimeSeriesId(object key1, object key2)
+        {
+            _key1 = (true, key1);
+            _key2 = (true, key2);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 3 keys.
+        /// </summary>
+        /// <param name="key1">The first key that identifies a Time Series instance.</param>
+        /// <param name="key2">The second key that identifies a Time Series instance.</param>
+        /// <param name="key3">The third key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 keys.
+        /// </remarks>
+        public TimeSeriesId(object key1, object key2, object key3)
+        {
+            _key1 = (true, key1);
+            _key2 = (true, key2);
+            _key3 = (true, key3);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 1 string key.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 string key.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1)
```

Comment:
rename param to timeSeriesIdStringProp1?

Reply:
Why?

Reply:
Binal, let's discuss offline on Monday. On one of the other comments earlier in the PR, we thought it is more appropriate to call them keys.

Reply:
Ok with key or property, I noticed it is differently named here than below and my comment was intended for it to have it consistent.